src/Image.php

```diff
@@ -163,11 +163,11 @@ protected function performOptimization($path, array $optimizerChainConfiguration
             $existingOptimizers = $optimizerChain->getOptimizers();
 
             $optimizers = array_map(function (array $optimizerOptions, string $optimizerClassName) use ($existingOptimizers) {
-                $optimizer = array_filter($existingOptimizers, function ($optimizer) use ($optimizerClassName) {
+                $optimizer = array_values(array_filter($existingOptimizers, function ($optimizer) use ($optimizerClassName) {
                     return get_class($optimizer) === $optimizerClassName;
-                });
+                }));
 
-                $optimizer = count($optimizer) > 0 && $optimizer[0] instanceof BaseOptimizer ? $optimizer[0] : new $optimizerClassName;
+                $optimizer = isset($optimizer[0]) && $optimizer[0] instanceof BaseOptimizer ? $optimizer[0] : new $optimizerClassName;
```

Comment:
Is this change necessary?

Reply:
It's safer.

```
php
$optimizer = [
    1 => (JpegOptim instance)
];
```

`count($optimizer) > 0` will be evaluated whereas `isset($optimizer[0])`. Even if it's not supposed to happen.